bundles/org.openhab.binding.gree/src/main/java/org/openhab/binding/gree/internal/handler/GreeAirDevice.java

```diff
@@ -117,9 +128,18 @@ public void getDeviceStatus(DatagramSocket clientSocket) throws GreeException {
             String reqStatusPackStr = GSON.toJson(reqStatusPackGson);
 
             // Encrypt and send the Status Request pack
-            String encryptedStatusReqPacket = GreeCryptoUtil.encryptPack(getKey(), reqStatusPackStr);
+            String encryptedStatusReqPacket;
+            String encryptedStatusReqTag = null;
+            if (encryptionType == EncryptionTypes.GCM) {
```

Comment:
could you move the logic to GreeCryptoUtil class, this layer here should not deal with it, just call a encrypt/decrypt method doing the details